poolscript/script.go

```diff
@@ -453,9 +453,11 @@ func IsExpirySpend(witness wire.TxWitness) bool {
 // IsTaprootExpirySpend determines whether the provided witness corresponds to
 // the expiration script path of a Taproot enabled (version 1) account.
 func IsTaprootExpirySpend(witness wire.TxWitness) bool {
-	// We still have 3 elements, the trader's signature, the script and the
-	// control block.
-	if len(witness) != 3 {
+	// At a minimum we expect these three elements: the trader's signature, the script, and the
+	// control block. We also accept an additional optional fourth element: the annex.
+	if hasAnnex(witness) && len(witness) != 4 {
+		return false
+	} else if len(witness) != 3 {
```

Comment:
nit: in Golang we usually try to avoid using the `else` statement if possible. That way the "happy path" is always to the left while error or special cases are indented.
It's all about readability, so I would probably write this block as follows:
```
if hasAnnex(witness) {
        // Snip off the last element that contains the annex.
        witness = witness[:len(witness)]
}

if len(witness) != 3 {
      return false
}
```

But of course that also comes down to personal taste.

Reply:
That's good to know, thanks. I've gone with your example.